### ISSUE TITLE
JSON component incorrectly handling dicts/strings with 'path' key

### ISSUE THREAD

**Issue by m4urin** [gradio-app/gradio#8367]
### Describe the bug

When a dictionary (or a string) is passed to the JSON component in Gradio, it modifies any nested dictionaries that contain the key 'path'.

Input example `test.json`:
```json
{
  "key1": 1, 
  "key2": {
    "key3": 3, 
    "path": "problem_here"
  }
}
```
Output by `gr.JSON():`
```json
{
  "key1": 1,
  "key2": {
    "path": "<path_to_reproduce_folder>\\problem_here",
    "url": "http://127.0.0.1:7860/file=<path_to_reproduce_folder>\\problem_here",
    "size": null,
    "orig_name": null,
    "mime_type": null,
    "is_stream": false,
    "meta": {
      "_type": "gradio.FileData"
    }
  }
}
```
This behavior is confusing and might not be intentional since the `load_json(file_path)` (see reproduction) function prints the data correctly and it is not affected by the gr.File() component. I can easily work around this issue by renaming the 'path' keys, but fixing this issue could prevent other users from having to spend time debugging this kind of output.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

Submit a json file with a 'path' key to this demo:
```python
import gradio as gr
import json

def load_json(file_path):
    with open(file_path, 'r') as file:
        data = json.load(file)
    return data

with gr.Blocks() as demo:
    input_file = gr.File(file_types=['.json'])
    button = gr.Button()
    output_file = gr.JSON()
    button.click(load_json, input_file, output_file)

if ___name__ == "__main__":
   demo.launch()
```

### Logs

```shell
Not available
```

### System Info

```shell
gradio==4.29.0	
gradio-client==0.16.1
```

### Severity

I can work around it

**Comment by freddyaboulton** [comment]
This has been fixed in version 4.31.3. Can you upgrade @m4urin ?

**Comment by m4urin** [comment]
@freddyaboulton, it works! I was working in the wrong environment and hadn't upgraded there yet.😊
